awscli/customizations/s3/utils.py

```diff
@@ -406,19 +407,39 @@ def uni_print(statement, out_file=None):
     out_file.flush()
 
 
-def bytes_print(statement):
+def bytes_print(statement, stdout=None):
     """
     This function is used to properly write bytes to standard out.
     """
+    if stdout is None:
+        stdout = sys.stdout
+
     if PY3:
```

Comment:
It would be nice to clean up some of this technical debt if we're in the code anyways, and move the py2/py3 differences over to compat.py.